### PR TITLE
[PM-5299] Extract a danger-zone component

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Extract a re-useable danger-zone component instead of needing to duplicate the code. I wasn't sure which team should own it, but defaulted to auth since ac is more an extension.

This results in a minimal change in the visual UI on the users account settings. But it matches the current styling of the AC which I consider acceptable.

## Screenshot

|Before|After|
|-|-|
|image|image|

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/web/src/app/auth/settings/account/danger-zone.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="tw-rounded tw-border tw-border-solid tw-border-danger-500 tw-p-5">
+  <p>{{ "dangerZoneDesc" | i18n }}</p>
+
+  <div class="tw-flex tw-flex-row tw-gap-2">
```
Comment (Hinton) [inline apps/web/src/app/auth/settings/account/danger-zone.component.html:4]: @willmartian I wonder if we should make a button-group component that has this spacing.

Comment (danielleflinn) [reply apps/web/src/app/auth/settings/account/danger-zone.component.html:4]: I like this idea, the button group component could also be used in the Dialog footer / form footers where we display a submit/cancel button if we wanted to keep spacing consistent.

Comment (willmartian) [reply apps/web/src/app/auth/settings/account/danger-zone.component.html:4]: Yes, 100%. 

Created a Jira issue: https://bitwarden.atlassian.net/browse/CL-158
Prior art: https://shoelace.style/components/button-group

**Review by danielleflinn — CHANGES_REQUESTED** [review]
Looks like the text in Dark themes is rendering wrong (dark gray) in Storybook. 
Screenshot 2023-12-14 at 8 07 36 AM

If we are keeping the component "Auth" owned, can we move it to the "Auth" collapsible section? 

Also what is the reasoning for keeping the "Danger zone" header out of the storybook component?

**Comment by Hinton** [comment]
@danielleflinn that's only in storybook. Looks fine in the actual app.

File: apps/web/src/app/auth/settings/account/danger-zone.component.html (hunk 1 above)
Comment (Hinton) [reply apps/web/src/app/auth/settings/account/danger-zone.component.html:4]: Shoelace is more of a toggle style. Our would just be the suggested gap between items. Which might mean it shouldn't be tied to button?

**Comment by Hinton** [comment]
I lifted in the title in the danger-zone component. We can revisit the decision later. This did make the header inconsistent so I updated the "Change email" to also use the new style.

File: apps/web/src/app/auth/settings/account/account.component.html [hunk 2]
```diff
@@ -2,29 +2,24 @@
   <h1>{{ "myAccount" | i18n }}</h1>
 </div>
 <app-profile></app-profile>
-<ng-container *ngIf="showChangeEmail">
-  <div class="secondary-header">
-    <h1>{{ "changeEmail" | i18n }}</h1>
-  </div>
+
+<div *ngIf="showChangeEmail" class="tw-mt-16">
+  <h1 bitTypography="h1">{{ "changeEmail" | i18n }}</h1>
```
Comment (willmartian) [inline apps/web/src/app/auth/settings/account/account.component.html:7]: For "Change Email", is removing the underline (was provided by `.secondary-header`) intended? cc @danielleflinn

**Review by willmartian — APPROVED** [review]
Good refactor. One question:

File: apps/web/src/app/auth/settings/account/account.component.html (hunk 2 above)
Comment (danielleflinn) [reply apps/web/src/app/auth/settings/account/account.component.html:7]: Yes, Design is looking to move away from the underlines shown below the headers

**Review by jlf0dev — APPROVED** [review]

**Review by danielleflinn — APPROVED** [review]

**Review by eliykat — APPROVED** [review]

**Pull request merged**